### PR TITLE
Add typings

### PR DESCRIPTION
**Pull request by endigo**
#14

### CODE REVIEW

**Comment by samwierema** [comment]
@endigo thanks for submitting a pull request! We'll have a look at this next week.

File: lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	lookup: {
+		read: (phoneNumber: string, countryCode: string, callback: Function) => {};
+		hlr: (phoneNumber: string, countryCode: string, callback: Function) => {};
+		create: (phoneNumber: string, params: any, callback: Function) => {};
+	}
+}
+
+declare function messagebird(accessKey: string, timeout?: number): MessageBird;
+
+export = messagebird;
```
Comment (epels) [inline lib/index.d.ts]: What do you think about also exporting the `MessageBird` interface?

```typescript
export {
    messagebird,
    MessageBird,
};
```

This takes more advantage of editors' auto-completion functionality.

File: lib/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		create: (recipients: any[], params: any, callback: Function) => {};
+	},
+	verify: {
+		read: (id: string, callback: Function) => {};
+		create: (recipient: number, params: any | Function, callback?: Function) => {};
+		delete: (id: string, callback: Function) => {};
+		verify: (id: string, token: string, callback: Function) => {};
+	},
+	lookup: {
+		read: (phoneNumber: string, countryCode: string, callback: Function) => {};
```
Comment (epels) [inline lib/index.d.ts]: Shouldn't this be:

```typescript
read: (phoneNumber: string, countryCode: string | Function, callback?: Function) => {};
```

For reference - implementation:
```javascript
read: function(phoneNumber, countryCode, callback) {
    var params = {};

    if (typeof countryCode === 'function') {
        callback = countryCode;
        countryCode = null;
    }

    if (countryCode) {
        params.countryCode = countryCode;
    }

    httpRequest('GET', '/lookup/' + phoneNumber + '/hlr', params, callback);
}
```

**Comment by miguelcobain** [comment]
Can this be merged?

**Comment by niekert** [comment]
Thanks a lot for laying the groundwork for the typings @endigo 🙌 

We decided to pick this up internally in #27 so we could tweak the TS setup a bit and add tests for the typings but your work in this PR is really appreciated :) 

Typings can be used onwards from `v2.3.2`

**Pull request closed without merging**